p2p/src/net/libp2p/mod.rs

```diff
@@ -447,4 +507,63 @@ mod tests {
             Err(P2pError::DecodeFailure("Message is too big".to_string()))
         );
     }
+
+    #[test]
+    fn test_parse_discovered_addr() {
+        let peer_id: PeerId =
+            "12D3KooWE3kBRAnn6jxZMdK1JMWx1iHtR1NKzXSRv5HLTmfD9u9c".parse().unwrap();
```

Comment:
This is probably ingrained into libp2p too much and almost certainly not worth the effort but I'd find bech32 peer IDs much nicer.

Reply:
bech32 would certainly be nicer but as you said, this is a libp2p thing and I think we're stuck with these.